hibernate-core/src/main/java/org/hibernate/persister/entity/JoinedSubclassEntityPersister.java

```diff
@@ -1394,7 +1412,12 @@ public void pruneForSubclasses(TableGroup tableGroup, Map<String, EntityNameUse>
 					tableReferenceJoins.add( join );
 				}
 				else {
-					final String tableExpression = oldJoin.getJoinedTableReference().getTableExpression();
+					final String tableExpression = joinedTableReference.getTableExpression();
+					final int index = indexOf( getSubclassTableNames(), tableExpression );
+					if ( index > -1 && isClassOrSuperclassTable[index] ) {
+						// Always retain joins to superclass tables
```

Comment:
Please explain why this is necessary. Would be a bummer if we can't remove unused super class table joins anymore.